schemas/questionnaire_v1.json

```diff
@@ -62,21 +62,37 @@
             "type": "string",
             "pattern": "^[a-zA-Z0-9-_]+$"
           },
-          "validator": {
+          "type": {
             "type": "string",
             "enum": [
               "boolean",
               "date",
               "string",
-              "optional_string",
-              "uuid"
+              "uuid",
+              "url"
             ]
+          },
+          "min_length": {
+            "type": "number",
+            "description": "The minimum length of a string field"
+          },
+          "max_length": {
+            "type": "number",
+            "description": "The maximum length of a string field"
+          },
+          "length": {
+            "type": "number",
+            "description": "The length of a string field must be equal to this"
+          },
+          "optional": {
```

Comment:
Since JSON and Marshmello both use required, would it be better if we also used `required` instead of `optional`?

Reply:
I prefer optional, since the field itself is optional. But happy to change if that's the consensus.

Coupling to marshmallow shouldn't really be an argument for it though